en/peripherals/frsky_telemetry.md

```diff
@@ -1,56 +1,209 @@
+* Turnigy 9XR Pro
+
+The transmitter must be bound to your receiver and be running customised scripts to display the telemetry (see following sections).
+
+### Taranis - LuaPilot Setup
+
+Compatible Taranis receivers (e.g. X9D Plus) running OpenTX 2.1.6 or newer can use the LuaPilot script to display received telemetry, as shown in the screenshot below.
+
+![Telemetry Screen on the Taranis](../../images/taranis_telemetry.jpg)
+
+Instructions for installing the script can be found here: [LuaPilot Taranis Telemetry script > Taranis Setup OpenTX 2.1.6 or newer](http://ilihack.github.io/LuaPilot_Taranis_Telemetry/)
+
+If you open the `LuaPil.lua` script with a text editor, you can edit the configuration. Suggested modifications include:
+* `local BattLevelmAh = -1` - Use the battery level calculation from the vehicle
+* `local SayFlightMode = 0` - There are no WAV files for the PX4 flight modes
+
+
+## Telemetry Messages {#messages}
+
+FrySky Telemetry can transmit most of the more useful status information from PX4. S-Port and D-Port receivers transmit different sets of messages, as listed the following sections.
+
+### S-Port {#s_port}
+
+S-Port receivers transmit the following messages (from [here](https://github.com/iNavFlight/inav/blob/master/docs/Telemetry.md#available-smartport-sport-sensors)):
+
+- **AccX, AccY, AccZ:** Accelerometer values.
+- **Alt:** Barometer based altitude, relative to home location.
+- **ASpd:** True air speed (from pitot sensor).
+- **A4:** Average cell value. 
+  > **Warning** Unlike FLVSS and MLVSS sensors, you do not get actual lowest value of a cell, but an average: `(total lipo voltage) / (number of cells)`
+- **Curr:** Actual current consumption (Amps).
+- **Fuel:** Remaining battery percentage if `battery_capacity` variable set and variable `smartport_fuel_percent = ON`, mAh drawn otherwise.
+- **GAlt:** GPS altitude, sea level is zero.
+- **GPS:** GPS coordinates.
+- **GSpd:** Current horizontal ground speed, calculated by GPS.
+- **Hdg:** Heading (degrees - North is 0°).
+- **VFAS:** Actual battery voltage value (Voltage FrSky Ampere Sensor).
+- **VSpd:** Vertical speed (cm/s).
+- **Tmp1:** Flight mode, sent as 5 digits. 
+
+  Number is sent as ABCDE detailed below. The numbers are additive (for example: if digit C is 6, it means both position hold and altitude hold are active).
    
-UART to S.PORT adapters can be sourced from:
+  - **A:** 1 = Flaperon mode, 2 = Auto tune mode, 4 = Failsafe mode.
+  - **B:** 1 = Return to home, 2 = waypoint mode, 4 = Headfree mode.
+  - **C:** 1 = Heading hold, 2 = Altitude hold, 4 = Position hold.
+  - **D:** 1 = Angle mode, 2 = Horizon mode, 4 = Passthrough mode.
+  - **E:** 1 = OK to arm, 2 = arming is prevented, 4 = armed.
+- **Tmp2:** GPS lock status, accuracy, home reset trigger, and number of satellites. Number is sent as ABCD detailed below. Typical minimum GPS 3D lock value is 3906 (GPS locked and home fixed, HDOP highest accuracy, 6 satellites).

```

Comment:
Tmp1 and Tmp2 are not standardized and have a different meaning in PX4:
- Tmp1 is the flight mode: https://github.com/PX4/Firmware/blob/master/src/drivers/telemetry/frsky_telemetry/frsky_telemetry.cpp#L85
- Tmp2: https://github.com/PX4/Firmware/blob/master/src/drivers/telemetry/frsky_telemetry/frsky_data.cpp#L236